### PR TITLE
ui: Make some dark-theme adjustments from Greg's feedback

### PR DESCRIPTION
**Pull request by chrisbobbe**
This addresses all the remaining items in the CZO thread at https://chat.zulip.org/#narrow/stream/48-mobile/topic/dark.20theme/near/1936853 .

Screenshots coming soon.

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Home page (light theme affected too because it was convenient to just swap a light/dark pair that comes from Material):

| Before | After |
| --- | --- |
| image | image |
| image | image |

Error style (light theme unchanged):

| Before | After |
| --- | --- |
| image | image |

Dot for muted unreads (light theme unchanged):

The way this looks, I think it might not be an improvement, actually. The color matches what gets used on web, but it ends up with pretty low contrast because the background color differs between mobile and web. But also, like many UI elements that currently follow web, this is something we'll revisit once we have a design from Vlad.

| Before | After |
| --- | --- |
| image | image |

**Comment by chrisbobbe** [comment]
cc @alya in case you'd like to take a look.

**Comment by alya** [comment]
Works for me, thanks!

File: lib/widgets/app.dart [hunk 1]
```diff
@@ -254,63 +256,68 @@ class HomePage extends StatelessWidget {
...
+            if (testStreamId != null) ...[
+              const SizedBox(height: 16),
+              ElevatedButton(
+                onPressed: () => Navigator.push(context,
+                  MessageListPage.buildRoute(context: context,
+                    narrow: ChannelNarrow(testStreamId!))),
+                child: const Text("#test here")), // scaffolding hack, see above
+            ],
+          ])),
+      ));
```
Comment (gnprice) [inline lib/widgets/app.dart:321]: nit:
```suggestion
          ]))));
```

File: lib/widgets/theme.dart [hunk 2]
```diff
@@ -158,6 +159,7 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
       groupDmConversationIconBg: const Color(0x33cccccc),
       loginOrDivider: const Color(0xff424242),
       loginOrDividerText: const Color(0xffa8a8a8),
+      mutedUnreadBadge: const HSLColor.fromAHSL(0.5, 0, 0, 0.3).toColor(),
```
Comment (gnprice) [inline lib/widgets/theme.dart:162]: This is pretty low-contrast in dark mode, but as with many of our UI
    elements, we'll adjust as necessary when we have a design from Vlad.

From the screenshot, this gets pretty hard to see. It's good to have this properly wired up into DesignVariables, but I think the existing value that's the same as in light theme actually works out better.

Let's maybe just make this the same value (with 0.8 lightness) as for the light theme, and leave a `// TODO(design-dark)` for it. Or spend a minute or two trying values in between like 0.6, and pick something that looks clear.

**Review by gnprice — COMMENTED** [review]
Thanks for taking care of these! Two small comments.

**Comment by chrisbobbe** [comment]
Thanks! Revision pushed.

Here's the before-and-after for `MutedUnreadBadge` in dark theme (`main` vs. the tip of the branch), with 0.6 for the lightness:

| Before | After |
| --- | --- |
| image | image |

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Pull request merged**